scim/usersRecords.bal

```diff
@@ -0,0 +1,454 @@
+// Copyright (c) 2023, WSO2 LLC. (http://www.wso2.com).
+
+// WSO2 LLC. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+
+// http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+# Represents the SCIM USER schema.
+#
+# + schemas - URIs of the used SCIM schemas  
+# + externalId - Identifier by the provisioning client  
+# + userName - Unique identifier within the system  
+# + name - Components of the user's name  
+# + displayName - Name displayed to end-users  
+# + nickName - Casual name of the user  
+# + profileUrl - URI of the user's online profile  
+# + emails - Email addresses for the User  
+# + addresses - Physical mailing address for the user  
+# + phoneNumbers - Phone numbers for the user  
+# + ims - Instant messaging addresses for the user  
+# + roles - List of roles for the user  
+# + photos - URI of image of the user  
+# + userType - Relationship between the organization and the user  
+# + title - Title of the user  
+# + preferredLanguage - Preferred written or spoken languages of the user  
+# + locale - Default location of the user  
+# + timezone - Time zone of the user  
+# + active - Administrative status of the user  
+# + password - Password of the user  
+# + groups - Groups to which the user belongs  
+# + x509Certificates - Certificates associated with the user  
+# + meta - Metadata of the user  
+# + urn\:ietf\:params\:scim\:schemas\:extension\:enterprise\:2\.0\:User - Attributes belong to business or enterprise extension schema  
+# + urn\:scim\:wso2\:schema - Attributes belong to WSO2 custom extension schema
+public type SCIMUser record {
+    string[]|string schemas?;
+    int externalId?;
+    string userName?;
+    Name name?;
+    string displayName?;
+    string nickName?;
+    string profileUrl?;
+    Email[] emails?;
+    Address[] addresses?;
+    Phone[] phoneNumbers?;
+    Ims[] ims?;
+    Role[] roles?;
+    Photo[] photos?;
+    string userType?;
+    string title?;
+    string preferredLanguage?;
+    string locale?;
+    string timezone?;
+    boolean active?;
+    string password?;
+    Group[] groups?;
+    Certificate[] x509Certificates?;
+    Meta meta?;
```

Comment:
Why https://www.rfc-editor.org/rfc/rfc7643.html#section-3.1:~:text=entitlements%0A%20%20%20%20%20%20A%20list,no%20canonical%20types. these two are not included?